File: Household Budget Calculator/[VB]-Household Budget Calculator/VB/Household budget calculator/expanding property classes/Insurance.vb
Imports System.ComponentModel

<TypeConverter(GetType(ExpandableObjectConverter))> <System.Serializable()> _
Public Class Insurance

    Public Event propertiesChanged(ByVal e As propertiesChangedEventArgs)

    Private _life As Decimal
    <DisplayName("Life + Protection"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property life() As Decimal
        Get
            Return _life
        End Get
        Set(ByVal value As Decimal)
            _life = value
            totalInsurance = life + motor + medical
        End Set
    End Property

    Private _motor As Decimal
    <DisplayName("Motor + Home"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property motor() As Decimal
        Get
            Return _motor
        End Get
        Set(ByVal value As Decimal)
            _motor = value
            totalInsurance = life + motor + medical
        End Set
    End Property

    Private _medical As Decimal
    <DisplayName("Medical"), _
    RefreshProperties(RefreshProperties.All)> _
    Public Property medical() As Decimal
        Get
            Return _medical
        End Get
        Set(ByVal value As Decimal)
            _medical = value
            totalInsurance = life + motor + medical
        End Set
    End Property

    Private _totalInsurance As Decimal
    <ReadOnlyAttribute(True), _
    Browsable(False), _
    DisplayName("Total Insurance Outgoings")> _
    Public Property totalInsurance() As Decimal
        Get
            Return _totalInsurance
        End Get
        Set(ByVal value As Decimal)
            _totalInsurance = value
            RaiseEvent propertiesChanged(New propertiesChangedEventArgs With {.propName = "Insurance", .newValue = value})
        End Set
    End Property

    Public Overrides Function ToString() As String
        Return totalInsurance.ToString("c2")
    End Function

End Class

File: Household Budget Calculator/[C#]-Household Budget Calculator/C#/Household budget calculator CS/expanding property classes/Insurance.cs
//using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.ComponentModel;

namespace Household_budget_calculator_CS.expanding_property_classes
{
    [TypeConverter(typeof(ExpandableObjectConverter))]
    [System.Serializable()]
    public class Insurance
    {
        public event propertiesChangedEventHandler propertiesChanged;
        public delegate void propertiesChangedEventHandler(propertiesChangedEventArgs e);

        private decimal _life;
        [DisplayName("Life + Protection"), RefreshProperties(RefreshProperties.All)]
        public decimal life
        {
            get { return _life; }
            set
            {
                _life = value;
                totalInsurance = life + motor + medical;
            }
        }

        private decimal _motor;
        [DisplayName("Motor + Home"), RefreshProperties(RefreshProperties.All)]
        public decimal motor
        {
            get { return _motor; }
            set
            {
                _motor = value;
                totalInsurance = life + motor + medical;
            }
        }

        private decimal _medical;
        [DisplayName("Medical"), RefreshProperties(RefreshProperties.All)]
        public decimal medical
        {
            get { return _medical; }
            set
            {
                _medical = value;
                totalInsurance = life + motor + medical;
            }
        }

        private decimal _totalInsurance;
        [ReadOnlyAttribute(true), Browsable(false), DisplayName("Total Insurance Outgoings")]
        public decimal totalInsurance
        {
            get { return _totalInsurance; }
            set
            {
                _totalInsurance = value;
                if (propertiesChanged != null)
                {
                    propertiesChanged(new propertiesChangedEventArgs
                    {
                        propName = "Insurance",
                        newValue = value
                    });
                }
            }
        }

        public override string ToString()
        {
            return totalInsurance.ToString("c2");
        }

    }
}
